changecalculator/src/main/java/org/msstate/cse/changecalculator/changeproneness/ChangePronenessManager.java

```diff
@@ -0,0 +1,154 @@
+package org.msstate.cse.changecalculator.changeproneness;
+
+
+import java.util.Collection;
+import java.util.Date;
+import java.util.Iterator;
+import java.util.List;
+
+import org.msstate.cse.changecalculator.database.DbUtil;
+import org.msstate.cse.changecalculator.diff.DiffParser;
+import org.msstate.cse.changecalculator.hibernate.HibernateUtil;
+import org.msstate.cse.changecalculator.hibernate.model.SvnHistroy;
+import org.msstate.cse.changecalculator.properties.Properties;
+import org.msstate.cse.changecalculator.svn.SvnUtil;
+import org.msstate.cse.changecalculator.utility.Util;
+import org.tmatesoft.svn.core.SVNException;
+import org.tmatesoft.svn.core.SVNLogEntry;
+import org.tmatesoft.svn.core.SVNURL;
+import org.tmatesoft.svn.core.auth.ISVNAuthenticationManager;
+import org.tmatesoft.svn.core.io.SVNRepository;
+import org.tmatesoft.svn.core.io.SVNRepositoryFactory;
+import org.tmatesoft.svn.core.wc.SVNWCUtil;
+
+public class ChangePronenessManager {
+	private String svnUrl = "";
+	private long startRevNum = -1;
+	private long endRevNum = -1;
+	
+	public ChangePronenessManager(Properties prop) {
+		if(null==prop){
+			System.err.println("Properties file is null");
+		}
+		this.svnUrl = prop.getSvn_url();
+		this.startRevNum = prop.getStartRevNum();
+		this.endRevNum = prop.getEndRevNum();
+	}
+	
+	public void calculateChangeProneness() {
+		if (Util.isNullOrEmpty(svnUrl)) {
+			System.err.println("Invalid SVN URL.");
+			return;
+		}
+		if (startRevNum < 0) {
+			System.err.println("Invalid start revision nubmer ");
+			return;
+		}
+//		if (endRevNum < 0) {
+//			System.err.println("Invalid end revision nubmer ");
+//			return;
+//		}
+
+		Collection<Long> logEntries = null;
+		try {
+			SVNRepository repository = getSvnRepository();
+			// end revision number
+			endRevNum = repository.getLatestRevision();
+			logEntries = repository.log(new String[] { "" }, null, startRevNum, endRevNum, true, true);
+		} catch (SVNException e) {
+			System.err.println(e.getMessage());
+			e.printStackTrace();
+			
+		}
+		
+		// ignore first revision
+		boolean isFirstRevision = true;
+		
+		//iterate over each log entries
+		for (Iterator entries = logEntries.iterator(); entries.hasNext();) {
+			try {

```

Comment:
The following code is ambiguous and may not be syntactically correct. SVNLogEntry appears to be a type/object, logEntry is the name, and (SVNLogEntry) is the parameter being passed to the new object, but then it is immediately followed by entries.next() not included as a parameter.